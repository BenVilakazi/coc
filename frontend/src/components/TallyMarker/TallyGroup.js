import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const propTypes = {
  color: PropTypes.oneOf(['primary', 'secondary']).isRequired,
  tallyCount: PropTypes.number.isRequired,
  faded: PropTypes.bool,
};

const StyledPath = styled.path`
  stroke: ${(props) => {
    if (props.faded) {
      return 'var(--accent-text-color)';
    }
    return props.color === 'primary'
      ? 'var(--primary-text-color)'
      : 'var(--secondary-text-color)';
  }};
`;

const tallyPaths = [
  'M32.4285 3.19775V29.5701',
  'M25.4439 3.51099L23.2217 29.5966',
  'M16.5548 3.51099L14.3326 29.5966',
  'M7.66589 3.51099L5.44367 29.5966',
  'M0.916168 0.599211L40.9161 29.1692',
];

function TallyGroup({ color, tallyCount, faded }) {
  return (
    <svg
      data-testid="tally-svg"
      width="42"
      height="30"
      viewBox="0 0 42 30"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        {tallyPaths.slice(0, tallyCount).map((tallyPath) => (
          <StyledPath
            key={tallyPath}
            color={color}
            faded={faded}
            d={tallyPath}
            strokeWidth="2"
            data-testid="tally"
          />
        ))}
      </g>
    </svg>
  );
}

TallyGroup.propTypes = propTypes;
TallyGroup.defaultProps = { faded: false };

export default TallyGroup;
