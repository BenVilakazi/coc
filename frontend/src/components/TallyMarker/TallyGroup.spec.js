import React from 'react';
import { render, screen } from '@testing-library/react';

import TallyGroup from './TallyGroup';

describe('TallyGroup', () => {
  it('renders', () => {
    render(<TallyGroup color="primary" tallyCount={3} />);
    expect(screen.getByTestId('tally-svg')).toBeInTheDocument();
  });

  describe('tallyCount', () => {
    it('renders all tallies', () => {
      render(<TallyGroup color="primary" tallyCount={5} />);
      const tallies = screen.getAllByTestId('tally');
      expect(tallies.length).toBe(5);
    });

    it('renders the expected amount of tallies', () => {
      render(<TallyGroup color="primary" tallyCount={2} />);
      const tallies = screen.getAllByTestId('tally');
      expect(tallies.length).toBe(2);
    });

    it('renders without any tallies', () => {
      render(<TallyGroup color="primary" tallyCount={0} />);
      const tallies = screen.queryAllByTestId('tally');
      expect(tallies.length).toBe(0);
    });
  });

  describe('color', () => {
    it('sets the color of the tallies to the primary color', () => {
      render(<TallyGroup color="primary" tallyCount={2} />);
      expect(screen.queryByTestId('tally-svg')).toMatchSnapshot();
    });

    it('sets the color of the tallies to the secondary color', () => {
      render(<TallyGroup color="secondary" tallyCount={2} />);
      expect(screen.queryByTestId('tally-svg')).toMatchSnapshot();
    });

    it('sets the color of the tallies to the faded color', () => {
      render(<TallyGroup color="secondary" tallyCount={2} faded />);
      expect(screen.queryByTestId('tally-svg')).toMatchSnapshot();
    });
  });
});
