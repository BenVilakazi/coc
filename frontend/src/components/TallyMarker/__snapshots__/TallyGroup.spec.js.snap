// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TallyGroup color sets the color of the tallies to the faded color 1`] = `
.c0 {
  stroke: var(--accent-text-color);
}

<svg
  data-testid="tally-svg"
  fill="none"
  height="30"
  viewBox="0 0 42 30"
  width="42"
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <path
      class="c0"
      color="secondary"
      d="M32.4285 3.19775V29.5701"
      data-testid="tally"
      stroke-width="2"
    />
    <path
      class="c0"
      color="secondary"
      d="M25.4439 3.51099L23.2217 29.5966"
      data-testid="tally"
      stroke-width="2"
    />
  </g>
</svg>
`;

exports[`TallyGroup color sets the color of the tallies to the primary color 1`] = `
.c0 {
  stroke: var(--primary-text-color);
}

<svg
  data-testid="tally-svg"
  fill="none"
  height="30"
  viewBox="0 0 42 30"
  width="42"
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <path
      class="c0"
      color="primary"
      d="M32.4285 3.19775V29.5701"
      data-testid="tally"
      stroke-width="2"
    />
    <path
      class="c0"
      color="primary"
      d="M25.4439 3.51099L23.2217 29.5966"
      data-testid="tally"
      stroke-width="2"
    />
  </g>
</svg>
`;

exports[`TallyGroup color sets the color of the tallies to the secondary color 1`] = `
.c0 {
  stroke: var(--secondary-text-color);
}

<svg
  data-testid="tally-svg"
  fill="none"
  height="30"
  viewBox="0 0 42 30"
  width="42"
  xmlns="http://www.w3.org/2000/svg"
>
  <g>
    <path
      class="c0"
      color="secondary"
      d="M32.4285 3.19775V29.5701"
      data-testid="tally"
      stroke-width="2"
    />
    <path
      class="c0"
      color="secondary"
      d="M25.4439 3.51099L23.2217 29.5966"
      data-testid="tally"
      stroke-width="2"
    />
  </g>
</svg>
`;
